app/database/server/models/channel.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+import {MM_TABLES} from '@constants/database';
+import field from '@nozbe/watermelondb/decorators/field';
+import children from '@nozbe/watermelondb/decorators/children';
+import Draft from '@typings/database/draft';
+import GroupsInChannel from '@typings/database/groups_in_channel';
+import PostInChannel from '@typings/database/post_in_channel';
+import MyChannelSettings from '@typings/database/my_channel_settings';
+import MyChannel from '@typings/database/my_channel';
+import ChannelInfo from '@typings/database/channel_info';
+import immutableRelation from '@nozbe/watermelondb/decorators/immutableRelation';
+import relation from '@nozbe/watermelondb/decorators/relation';
+import User from '@typings/database/user';
+import Team from '@typings/database/team';
+import ChannelMembership from '@typings/database/channel_membership';
+import Post from '@typings/database/post';
+
+export default class Channel extends Model {
+    static table = MM_TABLES.SERVER.CHANNEL
+    static associations: Associations = {
+        [MM_TABLES.SERVER.CHANNEL_INFO]: {type: 'has_many', foreignKey: 'channel_id'},
```

Comment:
this should be 1 to 1, basically this should not be part of  the association but a relation instead, something like

```js
@relation(MM_TABLES.SERVER.CHANNEL_INFO, 'channel_id') info;
```

Reply:
hey @enahum - I'm getting a confusion on this one.  I cross-checked the docs and the rn-example in the wdb repo. 

In a situation where a Parent and a Child models share a 1:1 relationship, the actual setup is to have, in the Parent entity, 'a has-many' association, while in the Child model, 'a belongs-to' association.  

In the child, we will have the "@relation(Parent, parent_id)" and in the Parent, we will have the '@children('child').  I've been trying the above scenario in the example app but it fails with 'blog.posts.observeCount' is undefined.

Can you please help me find out where I'm missing the point?